lib/screens/forgot_password_screen.dart

```diff
@@ -1,9 +1,136 @@
+
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
+import 'package:gezamycar/models/user_manager.dart';
+import 'package:gezamycar/utils/constants.dart';
+import 'package:gezamycar/widgets/custom_material_button.dart';
+import 'package:gezamycar/widgets/custom_text_form.dart';
+import '../widgets/custom_text_form.dart';
+import '../widgets/login_text_anim.dart';
+import 'package:gezamycar/utils/form_validators.dart';
+import 'package:gezamycar/common/myflutter_alert.dart';
+import 'package:rflutter_alert/rflutter_alert.dart';
+import 'login_screen.dart';
 
-class ForgotPasswordScreen extends StatelessWidget {
+class ForgotPasswordScreen extends StatefulWidget {
   static const String id = 'ForgotPasswordScreen';
+
+  @override
+  _ForgotPasswordScreen createState() => _ForgotPasswordScreen();
+}
+
+class _ForgotPasswordScreen extends State<ForgotPasswordScreen> {
+  final _formKey0 = GlobalKey<FormState>();
+  final resetAlert = MyFlutterAlert.instance;
+  String _email;
+
+  void _submitRestForm() async {
+    final form = _formKey0.currentState;
+    final _manager = UserManager.instance;
+
+    if (form.validate()) {
+      print(_email);
+      _manager.resetPassword(_email);
```

Comment:
Wrap this call inside a try-catch block. This way we will be able to let the user know if they have the reset link sent to their email or the provided wrong email address.